litho-core/src/main/java/com/facebook/litho/DrawableComponent.java

```diff
@@ -107,6 +108,26 @@ private void setDrawableWidth(int drawableWidth) {
     mDrawableWidth = drawableWidth;
   }
 
+  /**
+   * Indicates if this component is a background for a ComponentHost.
+   *
+   * <p>This will help to check if the background is a ripple drawable. We have to set
+   * RippleDrawables to the view's background so ripple can be projected outside the view bounds.
+   */
+  public void setIsBackground(boolean isBackground) {
+    this.isBackground = isBackground;
+  }
+
+  /**
+   * Returns true if this component is a background for a ComponentHost or else returns false.
+   *
+   * <p>This will help to check if the background is a ripple drawable. We have to set
+   * RippleDrawables to the view's background so ripple can be projected outside the view bounds.
+   */
+  public boolean isBackground() {
+    return isBackground;
+  }
+
```

Comment:
How about moving most of Ripple Background checks here? In such way you don't need to do checks twice (during mount and during unmount) There could be two methods like
```
public void maybeSetIsRippleBackground() {
   if (SDK_INT < VERSION_CODES.LOLLIPOP) {
      return;
    }

    if (!(mDrawable instanceof DefaultComparableDrawable)) {
      return;
    }
    // We need to unwrap the RippleDrawable from DefaultComparableDrawable since
    // DefaultComparableDrawable does not override all the methods necessary for Ripple drawable to
    // be projected outside the bounds.
    // We need to get the wrapped drawable to identify if it's a ripple drawable or not.
    final Drawable unwrappedDrawable = ((DefaultComparableDrawable) mDrawable).getWrappedDrawable();
    // We will set the view's background only if it's RippleDrawable
    if (unwrappedDrawable instanceof RippleDrawable) {
      isRippleBackground = true;
    }
}

public boolean isRippleBackground() {
    return isRippleBackground;
}
```

And `ComponentHost` code would be much simpler with this. Just checking `DrawableComponent#isRippleBackground` should be enough to decide whether to set bg or not.

Reply:
1. We can move this method inside the DrawableComponent, but we still need to do some operations for getting the RippleDrawable from MountItem. 
2. Since getDrawable() method of DrawableComponent is private, we have do unwrap the drawable from MountItem in ComponentHost and get the Ripple Drawable. Right?
Please suggest.

Reply:
1. Oh yeah, that's fair.
2. You could change getDrawable to be package-private.

Reply:
ok, I will try that and update you here. Thanks for your review.

Reply:
Done. Please review.